### PR TITLE
Bluetooth Discovery & Broadcaster minor changes

### PR DESCRIPTION
**Pull request by ftyl**
Hello,

started looking at the code a little and stumbled over a few things... I am not sure about the usefulness of my changes in BluetoothBroadcastService.swift, so maybe these should be ignored, however I have been taught to try to avoid force-unwrapping of variables even if it seems extremely safe. (line 77 and 141).

BluetoothDiscoveryService.swift -- Changes here are true concerns about race conditions and memory leaks, even if unlikely to occur, but if bluetooth gets turned on and off for example while the app is running, this, I believe, really can occur.

### CODE REVIEW

File: Sources/DP3TSDK/Bluetooth/BluetoothBroadcastService.swift [hunk 1]
```diff
@@ -67,18 +67,27 @@ class BluetoothBroadcastService: NSObject {
...
-        let characteristic = CBMutableCharacteristic(type: Default.shared.parameters.bluetooth.characteristicsCBUUID,
-                                                     properties: [.read, .notify],
-                                                     value: nil,
-                                                     permissions: .readable)
-        service?.characteristics = [characteristic]
-        peripheralManager?.add(service!)
+        service = CBMutableService(type: Default.shared.parameters.bluetooth.serviceCBUUID, primary: true)
+        
+        /// make sure creation of the service actually worked and log during calibration if not, otherwise force-unwrapping service will crash the app
+        if let service = service {
```
Comment (stmitt) [inline Sources/DP3TSDK/Bluetooth/BluetoothBroadcastService.swift:73]: The init of a CBMutableService will always succeed[1] so i think there is no need to introduce a new error type.
Were you able to produce this kind of behaviour?

[1] https://developer.apple.com/documentation/corebluetooth/cbmutableservice/1434330-init

Comment (ftyl) [reply Sources/DP3TSDK/Bluetooth/BluetoothBroadcastService.swift:73]: This is what I expected, yes. As I said, my changes in BluetoothBroadcastService are not as important or logical...

**Comment by ftyl** [comment]
Sorry, I messed up the whole branch due to some Xcode mess-up with symlinks which forced me to remove CALIBRATION on my end to test :/

File: Sources/DP3TSDK/DP3TError.swift [hunk 2]
```diff
@@ -25,6 +25,9 @@ public enum DP3TTracingError: Error {
 
     /// Bluetooth device turned off
     case bluetoothTurnedOff
+    
+    /// Bluetooth service setup failed
+    case bluetoothServiceFailed
```
Comment (ftyl) [inline Sources/DP3TSDK/DP3TError.swift:30]: Due to changes in my modification of the Broadcaster this is really not needed anymore.

**Comment by stmitt** [comment]
We recently undertook a massive refactoring to integrate Apple's official ExposureNotification framework in favour of our custom Bluetooth and Key-generation implementation. Going forward, we will only support the official Apple framework that handles all bluetooth communication. Please update to the latest SDK version and subscribe to the notifications so you keep up to date with the latest changes.

**Pull request closed without merging**